### ISSUE TITLE
Add Optionals 

### ISSUE THREAD

**Issue by knjk04** [Project-Books/book-project#356]
Where a null is used as a return type, change this to use an `Optional` instead. We don't want to use this for parameters

This is in line with a change to our style guide.

Branch: `0.2.0`

**Comment by luisa01-code** [comment]
Hi! I could fix that if you want

**Comment by knjk04** [comment]
Sure, thanks!

**Comment by raulsoledispa** [comment]
Hi, I would like to can help you. Can you assign me it?

**Comment by knjk04** [comment]
@raulsoledispa Hi, this issue is already assigned, unfortunately. Feel free to request to be assigned to an unassigned and unblocked issue

**Comment by knjk04** [comment]
@luisa01-code One thing I forgot to mention: let us know over [Slack](https://join.slack.com/t/teambookproject/shared_invite/zt-h6vx3n8l-KN8QnO50r7QWZHgHFnFdPw) if you need a hand with this

**Comment by knjk04** [comment]
@raulsoledispa This issue is now available again. Are you still interested in working on this?

**Comment by raulsoledispa** [comment]
@knjk04  Yes please

**Comment by knjk04** [comment]
This issue has been offered to someone on Slack. If they don't wish to pick this up, this will be available to everyone else

**Comment by hieu-t-tran** [comment]
I would like to work on this issue.

**Comment by knjk04** [comment]
@hieu-t-tran Thanks! Also, we're not looking to add this for fields. Raul got quite far, so you may wish to look at their PR if you get stuck

**Comment by knjk04** [comment]
@hieu-t-tran How are you getting on with this issue? If you need any help on this, feel free to reach out to us on [Slack](https://join.slack.com/t/teambookproject/shared_invite/zt-i7lept44-rgJ9yB0A2vedJTLyyfkjKQ)

If you need more time, that's fine, but please let us know.

**Comment by hieu-t-tran** [comment]
@knjk04 Thanks for checking up! I'm still working on this issue, and I hope to make a pull request by Tuesday next week if possible. I should have more time this weekend to work on it.

**Comment by knjk04** [comment]
@hieu-t-tran Great, thanks for the update!

**Comment by knjk04** [comment]
@hieu-t-tran Just checking up again -- do you have an update on this?

**Comment by shivansh29** [comment]
Hey is this issue available??

**Comment by knjk04** [comment]
Hi @shivansh29, I'm afraid not. Hieu is working on this

**Comment by knjk04** [comment]
@hieu-t-tran I've reassigned you. I've also updated the issue description to say that we should target `0.2.0`, as discussed. Could you please, therefore, branch off `0.2.0` and, when you're ready, create a PR to go back into it?

**Comment by hieu-t-tran** [comment]
Just as an update, I have finished installing the 0.2.0 branch and am in the process of bringing all of the changes from my previous pull request to a new branch. I will make a new pull request after this new branch is in sync with my previous pull request and then keep working on the changes you requested.
